src/plumpy/process_states.py

```diff
@@ -52,16 +52,17 @@ class Interruption(Exception):  # noqa: N818
 
 
 class KillInterruption(Interruption):
-    def __init__(self, msg_text: str | None):
-        super().__init__()
-        msg = MessageBuilder.kill(text=msg_text)
+    def __init__(self, msg_text: str | None, force_kill: bool = False):
+        super().__init__(msg_text)
+        msg = MessageBuilder.kill(text=msg_text, force_kill=force_kill)
 
         self.msg: MessageType = msg
+        self.force_kill: bool = force_kill
 
 
 class PauseInterruption(Interruption):
     def __init__(self, msg_text: str | None):
-        super().__init__()
+        super().__init__(msg_text)
```

Comment:
this is just to include the msg_text in the cancel exception, improves readability

Reply:
I don't think the `Interrupution` class is needed in the first place. Can `KillInterruption` and PauseInterruption` directly subclass of `Exception`? Then this `super()` call can be avoided and the instantiate of the interruption exception get more straightforward.

Reply:
Well I don't want to try out in this PR any changes to the Interruption class structure. I just added here the msg_text for better debugging. I would move this to an issue if this is important.

Reply:
Make sense.